src/App.js

```diff
@@ -22,19 +23,19 @@ function App() {
           render={({ data }) => {
             return (
               <div className="App">
-                <Route
-                  exact
-                  path="/"
-                  render={() => <Shopping list={data} userToken={userToken} />}
-                />
-                <Route
-                  path="/add"
-                  render={() => <AddItem userToken={userToken} list={data} />}
-                />
-                <nav id="nav">
-                  <NavButton path="/" text="Shopping" />
-                  <NavButton path="/add" text="Add Item" />
-                </nav>
+                <Switch>
+                  <Route
+                    exact
+                    path="/:itemId?"
+                    render={() => (
+                      <Shopping userToken={userToken} list={data} />
+                    )}
+                  />
+                  <Route
```

Comment:
The `/add` route needs to go first @espain16 @skillitzimberg. Switch will match the first path that it matches.